pkg/lib/registry/registry.go

```diff
@@ -228,3 +228,89 @@ func (r RegistryUpdater) PruneFromRegistry(request PruneFromRegistryRequest) err
 
 	return nil
 }
+
+type DeprecateFromRegistryRequest struct {
+	Permissive    bool
+	SkipTLS       bool
+	CaFile        string
+	InputDatabase string
+	Bundles       []string
+	ContainerTool containertools.ContainerTool
+}
+
+func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequest) error {
+	db, err := sql.Open("sqlite3", request.InputDatabase)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	dbLoader, err := sqlite.NewSQLLiteLoader(db)
+	if err != nil {
+		return err
+	}
+	if err := dbLoader.Migrate(context.TODO()); err != nil {
+		return fmt.Errorf("hello 3 %s", err)
+	}
+	dbQuerier := sqlite.NewSQLLiteQuerierFromDb(db)
+
+	// add custom ca certs to resolver
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil || rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+	if len(request.CaFile) > 0 {
+		certs, err := ioutil.ReadFile(request.CaFile)
+		if err != nil {
+			return fmt.Errorf("failed to append %q to RootCAs: %v", certs, err)
+		}
+		if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+			return fmt.Errorf("unable to add certs specified in %s", request.CaFile)
+		}
+	}
+
+	var reg image.Registry
+	var rerr error
+	switch request.ContainerTool {
+	case containertools.NoneTool:
+		reg, rerr = containerdregistry.NewRegistry(containerdregistry.SkipTLS(request.SkipTLS), containerdregistry.WithRootCAs(rootCAs))
+	case containertools.PodmanTool:
+		fallthrough
+	case containertools.DockerTool:
+		reg, rerr = execregistry.NewRegistry(request.ContainerTool, r.Logger)
+	}
+	if rerr != nil {
+		return rerr
+	}
+	defer func() {
+		if err := reg.Destroy(); err != nil {
+			r.Logger.WithError(err).Warn("error destroying local cache")
+		}
+	}()
+
+	for _, bundlePath := range request.Bundles {
+		csvName, err := dbQuerier.GetCSVNameForImage(context.TODO(), bundlePath)
+		if err != nil || csvName == "" {
+			r.Logger.WithError(err).Warn("error getting csv for bundle ", bundlePath)
+			continue
+		}
+		version, err := dbQuerier.GetBundleVersion(context.TODO(), bundlePath)
+		if err != nil {
+			r.Logger.WithError(err).Warn("error getting version for bundle ", bundlePath)
+			continue
+		}
+		tailBundles, err := dbQuerier.GetTailForBundleInChannel(context.TODO(), csvName)
```

Comment:
I think we want this query to be based on the `(bundle, version, bundlepath)` to help future proof for a time when csvName is not unique